### PR TITLE
Rst to markdown

### PR DESCRIPTION
**Pull request by larzuk91**
Solving #59

### CODE REVIEW

File: knpu/client-credentials.rst [hunk 1]
```diff
@@ -20,13 +19,12 @@ So COOP is real, sort of. You can find this make-believe website by going
...
-Starting our Command-line Script
---------------------------------
+## Starting our Command-line Script
 
 COOP's API is simple, with just a few endpoints, including the one we
 want for our little command-line script: eggs-collect.
 
-I've already made a ``cron/`` directory with a script called ``collect_eggs.php``
+I've already made a `cron/` directory with a script called `collect_eggs.php`
 that'll get us started::
```
Comment (bocharsky-bw) [inline knpu/client-credentials.rst]: Double colon `::` is an .rst feature, so we should replace all `::` at the end with `:` and use \`\`\` for next code block.

File: knpu/client-credentials.rst [hunk 2]
```diff
@@ -20,13 +19,12 @@ So COOP is real, sort of. You can find this make-believe website by going
...
 
     // collect_eggs.php
```
Comment (bocharsky-bw) [inline knpu/client-credentials.rst]: in this case we should use \`\`\`php since we have PHP code here

File: knpu/client-credentials.rst [hunk 3]
```diff
@@ -40,29 +38,29 @@ that'll get us started::
...
-into the ``cron/`` directory and then install the vendor libraries:
-
-.. code-block:: bash
-
-    php composer.phar install
+Before we start, we need to use [Composer][Composer] to download Guzzle. [Download Composer][Download Composer]
+into the `cron/` directory and then install the vendor libraries:
 
-.. note::
+```bash
```
Comment (bocharsky-bw) [inline knpu/client-credentials.rst]: Let's use `terminal` here too

File: knpu/client-credentials.rst [hunk 4]
```diff
@@ -77,21 +75,20 @@ number will be different::
 
 Try it by executing the script from the command line:
 
-.. code-block:: bash
-
-    php collect_eggs.php
+```bash
```
Comment (bocharsky-bw) [inline knpu/client-credentials.rst]: `terminal`

File: knpu/client-credentials.rst [hunk 5]
```diff
@@ -222,8 +215,8 @@ docs, we can see the URL and the POST parameters it needs:
         client_secret
         grant_type
 
-Let's update our script to first make *this* API request. Fill in the ``client_id``,
-``client_secret`` and ``grant_type`` POST parameters::
+Let's update our script to first make *this* API request. Fill in the `client_id`,
+`client_secret` and `grant_type` POST parameters::
```
Comment (bocharsky-bw) [inline knpu/client-credentials.rst]: `::` -> `:`

File: knpu/facebook.rst [hunk 6]
```diff
@@ -234,17 +227,16 @@ right, and it can be done with the help of the SDK::
 When we try the process again, we get a valid-looking user id. So, what just
 happened?
 
-The ``getUser`` method does a whole lot more than it looks like. It actually
-looks for the ``code`` query parameter and makes the API request to get the 
+The `getUser` method does a whole lot more than it looks like. It actually
```
Comment (bocharsky-bw) [inline knpu/facebook.rst]: Let's add `()` to all method names. If you edit a page and accidentally stumble on method or function without `()` - just add it! It's not much important, so don't look for this specifically ;)

File: knpu/facebook.rst [hunk 7]
```diff
@@ -234,17 +227,16 @@ right, and it can be done with the help of the SDK::
...
 can keep in mind how OAuth works and what's happening behind the scenes at
 each step, you'll be in great shape when something goes wrong.
 
-Handling Failure
-----------------
+## Handling Failure
 
 Just like with COOP, we need to handle failure. If we're missing the authorization
-code or something else goes wrong behind the scenes, the ``getUser`` method
+code or something else goes wrong behind the scenes, the `getUser` method
```
Comment (bocharsky-bw) [inline knpu/facebook.rst]: `getUser()`

File: knpu/facebook.rst [hunk 8]
```diff
@@ -273,14 +265,14 @@ account. Unlike COOP, most OAuth servers remember if you authorized an app
...
-``error_reason`` query parameters. But instead of seeing the error template,
+request. After the redirect, we see the `error`, `error_description` and
+`error_reason` query parameters. But instead of seeing the error template,
 our valid userId is printed out as if it were successful. What just happened?
 
 Our OAuth flow *did* fail. But even still, the Facebook object looks and
 finds a valid access token that it stored in the session from the last, successful
 authorization. That's nice, but it's unexpected. Just remember that
-``getUser`` tries many things: like exchanging the authorization code for
+`getUser` tries many things: like exchanging the authorization code for
```
Comment (bocharsky-bw) [inline knpu/facebook.rst]: `getUser()`

File: knpu/facebook2.rst [hunk 9]
```diff
@@ -1,27 +1,25 @@
...
+        Share how many eggs you've collected today on Facebook!
+        <a href="{{ path('facebook_share_place') }}" class="btn btn-info">Share</a>
+    {% else %}
+        Share your status on Facebook!
+        <a href="{{ path('facebook_authorize_start') }}">Connect with Facebook</a>
+    {% endif %}
+</div>
+```
+
+The URL I'm generating here is pointing to a function called `shareProgressOnFacebook`
```
Comment (bocharsky-bw) [inline knpu/facebook2.rst]: `shareProgressOnFacebook()`

File: knpu/facebook2.rst [hunk 10]
```diff
@@ -1,27 +1,25 @@
...
 in FacebookOAuthController::
```
Comment (bocharsky-bw) [inline knpu/facebook2.rst]: `::` -> `:`

File: knpu/facebook2.rst [hunk 11]
```diff
@@ -1,27 +1,25 @@
...
 
     // src/OAuth2Demo/Client/Controllers/FacebookOAuthController.php
```
Comment (bocharsky-bw) [inline knpu/facebook2.rst]: \`\`\`php

File: knpu/facebook2.rst [hunk 12]
```diff
@@ -34,23 +32,22 @@ in FacebookOAuthController::
...
+`/[USER_ID]/feed` and send `message` and `access_token` POST data.
 
 We could *absolutely* do this manually, using the nice Guzzle library from
 before. But since we're using the Facebook SDK, it's even easier.
 
-Use the ``createFacebook`` method from before to get our Facebook object
-and then use its ``api`` method. This takes 3 arguments: the API URL, the
+Use the `createFacebook` method from before to get our Facebook object
+and then use its `api` method. This takes 3 arguments: the API URL, the
 HTTP method, and any parameters we need to send::
```
Comment (bocharsky-bw) [inline knpu/facebook2.rst]: `:`

File: knpu/facebook2.rst [hunk 13]
```diff
@@ -34,23 +32,22 @@ in FacebookOAuthController::
...
 
     public function shareProgressOnFacebook()
```
Comment (bocharsky-bw) [inline knpu/facebook2.rst]: \`\`\`php

File: knpu/facebook2.rst [hunk 14]
```diff
@@ -123,12 +120,11 @@ the flow by redirecting back to the homepage::
...
 
-Handling Failure and Re-Authorizing
------------------------------------
+## Handling Failure and Re-Authorizing
 
 Of course, the API request may fail, especially in the world of OAuth where
 the access token might be expired. If any API request fails, the Facebook
-class will throw a ``FacebookApiException``. That's great, because
+class will throw a `FacebookApiException`. That's great, because
 we can wrap the API call in a try-catch block::
```
Comment (bocharsky-bw) [inline knpu/facebook2.rst]: `:`

File: knpu/facebook2.rst [hunk 15]
```diff
@@ -123,12 +120,11 @@ the flow by redirecting back to the homepage::
...
 
     try {
```
Comment (bocharsky-bw) [inline knpu/facebook2.rst]: \`\`\`php

File: knpu/facebook2.rst [hunk 16]
```diff
@@ -144,10 +140,10 @@ we can wrap the API call in a try-catch block::
     }
 
 If you want to get information about the error, the exception object has
-a few useful methods, like ``getResult``, which gives you the raw API error
-response or ``getType`` and ``getCode``. Facebook has a helpful page called
```
Comment (bocharsky-bw) [inline knpu/facebook2.rst]: `getResult()`, `getType()` and `getCode()` - just a good chance to fix it :)

File: knpu/facebook2.rst [hunk 17]
```diff
@@ -144,10 +140,10 @@ we can wrap the API call in a try-catch block::
...
-`Using the Graph API`_ that talks about the API and also the errors you might
-get back. If ``getType`` returns ``OAuthException``, or if the code is
+a few useful methods, like `getResult`, which gives you the raw API error
+response or `getType` and `getCode`. Facebook has a helpful page called
+[Using the Graph API][Using the Graph API] that talks about the API and also the errors you might
+get back. If `getType` returns `OAuthException`, or if the code is
```
Comment (bocharsky-bw) [inline knpu/facebook2.rst]: `getType()`

File: knpu/facebook2.rst [hunk 18]
```diff
@@ -144,10 +140,10 @@ we can wrap the API call in a try-catch block::
...
 190 or 102, the error is probably related to OAuth and we should try 
 re-authorizing them::
```
Comment (bocharsky-bw) [inline knpu/facebook2.rst]: `:` and a php code block on the next line

File: knpu/facebook2.rst [hunk 19]
```diff
@@ -182,11 +178,10 @@ API requests fail due to an expired access token, you can give your users
 a better experience by having them re-authorize your application instead
 of just failing.
 
-Re-trying an API Request
-~~~~~~~~~~~~~~~~~~~~~~~~
+### Re-trying an API Request
 
 Depending on the error, you might also want to re-try the request. Let's
-refactor the API call into a new private method called ``makeApiRequest``::
+refactor the API call into a new private method called `makeApiRequest`::
```
Comment (bocharsky-bw) [inline knpu/facebook2.rst]: `makeApiRequest()`

File: knpu/facebook2.rst [hunk 20]
```diff
@@ -182,11 +178,10 @@ API requests fail due to an expired access token, you can give your users
...
 
     public function shareProgressOnFacebook()
```
Comment (bocharsky-bw) [inline knpu/facebook2.rst]: php code block

File: knpu/facebook2.rst [hunk 21]
```diff
@@ -226,11 +221,11 @@ refactor the API call into a new private method called ``makeApiRequest``::
         }
     }
 
-This method does the exact same thing as before. The ``if`` statement checks
-to see if ``makeApiRequest`` needs us to redirect the user back to the authorize
+This method does the exact same thing as before. The `if` statement checks
+to see if `makeApiRequest` needs us to redirect the user back to the authorize
```
Comment (bocharsky-bw) [inline knpu/facebook2.rst]: `makeApiRequest()`

File: knpu/facebook2.rst [hunk 22]
```diff
@@ -226,11 +221,11 @@ refactor the API call into a new private method called ``makeApiRequest``::
...
 URL.
 
-But if we add a new ``$retry`` argument, we could run the request 1 more time if it fails::
+But if we add a new `$retry` argument, we could run the request 1 more time if it fails::
```
Comment (bocharsky-bw) [inline knpu/facebook2.rst]: `:` and php code block the next

File: knpu/facebook2.rst [hunk 23]
```diff
@@ -377,7 +371,7 @@ into our FacebookOAuthController::
 But to create a user, we need some basic information, like email, first name
 and last name. With COOP, we made an API request to get this information.
 Let's do the same thing for Facebook, using the really important endpoint
-``/me``. And knowing that things can fail, let's make sure to wrap it in
+`/me`. And knowing that things can fail, let's make sure to wrap it in
 a try-catch block::
```
Comment (bocharsky-bw) [inline knpu/facebook2.rst]: `:` and php code block the next

File: knpu/facebook2.rst [hunk 24]
```diff
@@ -405,28 +399,28 @@ redirecting them to re-authorize.
 I'm dumping the result of the API request, so let's logout and try the process. 
 But first, reset the database so that it doesn't find our existing user:
 
-.. code-block:: bash
-
-    rm data/topcluck.sqlite
+```bash
```
Comment (bocharsky-bw) [inline knpu/facebook2.rst]: `terminal`

File: knpu/facebook2.rst [hunk 25]
```diff
@@ -405,28 +399,28 @@ redirecting them to re-authorize.
...
 We're allowed to ask for this information because when we redirect the user
-for authorization, we're asking for the ``email`` scope. Let's update the
-``findOrCreateUser`` method to use this data.
+for authorization, we're asking for the `email` scope. Let's update the
+`findOrCreateUser` method to use this data.
 
-First, change ``findUserByCOOPId`` to ``findUserByFacebookId``, which is
-a shortcut method in my app to find a user by the  ``facebookUserId`` column::
+First, change `findUserByCOOPId` to `findUserByFacebookId`, which is
+a shortcut method in my app to find a user by the  `facebookUserId` column::
```
Comment (bocharsky-bw) [inline knpu/facebook2.rst]: `findOrCreateUser()`, `findUserByCOOPId()` and `findUserByFacebookId()` with `:` at the end and php code block on the next line

File: knpu/implicit2.rst [hunk 26]
```diff
@@ -1,118 +1,115 @@
...
-    <div id="google-plus-farmers"></div>
-    <!-- ... -->
+<!-- ... -->
+<a href="#" class="btn btn-lg btn-info js-google-signin">Connect with Google+</a>
+<div id="google-plus-farmers"></div>
+<!-- ... -->
+```
 
 Let's call this function automatically after we authenticate. This code loads the
 Google+ part of the SDK and calls our function.:
```
Comment (bocharsky-bw) [inline knpu/implicit2.rst]: Let's fix the mistake which was before you: `.:` -> `:`

File: knpu/implicit2.rst [hunk 27]
```diff
@@ -1,118 +1,115 @@
...
+## Page parameters
 
 Our ultimate goal is for the user to be able to choose from the people in
 their circles and invite them to join TopCluck. With all the OAuth stuff behind
 us, this is just a matter of writing some JavaScript and figuring out exactly
 how to use the Google+ API to accomplish this. We'll leave this to you!
 
 But there's one more small thing that's bothering me. When we click to sign in,
-the sign-in function is called twice, which means ``loadCirclesPeople``
+the sign-in function is called twice, which means `loadCirclesPeople`
```
Comment (bocharsky-bw) [inline knpu/implicit2.rst]: `loadCirclesPeople()`

File: knpu/logging-in.rst [hunk 28]
```diff
@@ -12,10 +11,9 @@ have a COOP account.
...
 COOP" or "Login with Facebook" buttons is really easy.
 
-Creating New TopCluck Users
----------------------------
+## Creating New TopCluck Users
 
-Start back in ``CoopOAuthController.php``, where we handled the exchange of the
+Start back in `CoopOAuthController.php`, where we handled the exchange of the
 authorization code for the access token. Right now, this assumes that
 the user is already logged in and updates their account with the COOP details::
```
Comment (bocharsky-bw) [inline knpu/logging-in.rst]: `:` and php code block

File: knpu/logging-in.rst [hunk 29]
```diff
@@ -111,47 +108,45 @@ Finally, let's log the user into this new account::
...
 not, we'll create a new account and log you in. It's that simple!
 
 Let's also completely reset the database, which you can do just by deleting
-the ``data/topcluck.sqlite`` file inside the ``client/`` directory:
-
-.. code-block:: bash
+the `data/topcluck.sqlite` file inside the `client/` directory:
 
-    $ rm data/topcluck.sqlite
+```bash
```
Comment (bocharsky-bw) [inline knpu/logging-in.rst]: `terminal`

File: knpu/refresh-token.rst [hunk 30]
```diff
@@ -26,8 +24,8 @@ gains a refresh token, it is useless to them without the client's credentials,
 as you'll see. Having two keys instead of one is a method often used in security 
 to make it harder for attackers to compromise a system.
 
-Fortunately, COOP *does* support refresh tokens. Open up the ``CoopOAuthController``
-where we make the API request to ``/token``. Let's dump this response and
+Fortunately, COOP *does* support refresh tokens. Open up the `CoopOAuthController`
+where we make the API request to `/token`. Let's dump this response and
 go through the process::
```
Comment (bocharsky-bw) [inline knpu/refresh-token.rst]: `:` and php code block

File: knpu/refresh-token.rst [hunk 31]
```diff
@@ -102,33 +99,32 @@ right at the top of the code that handles the redirect::
...
 
 I've already started the little script for the CRON job, which you can see
-at ``data/refresh_tokens.php``. What we want to do here is use the COOP API
+at `data/refresh_tokens.php`. What we want to do here is use the COOP API
 to count and save each user's daily eggs.
 
 But first, we need to make sure that everyone has a non-expired access token.
-Let's use a method called ``getExpiringTokens`` that I've already prepared.
-This queries the database and returns details for all users whose ``coopAccessExpiresAt``
+Let's use a method called `getExpiringTokens` that I've already prepared.
```
Comment (bocharsky-bw) [inline knpu/refresh-token.rst]: `getExpiringTokens()`

File: knpu/refresh-token.rst [hunk 32]
```diff
@@ -148,17 +144,17 @@ value is today or earlier::
...
-    In the background, this is just running a query similar to this:
+***TIP
+In the background, this is just running a query similar to this:
+***
 
-    .. code-block:: text
-
-        SELECT * FROM users WHERE coopAccessExpiresAt < '2014-XX-YY';
+```text
+    SELECT * FROM users WHERE coopAccessExpiresAt < '2014-XX-YY';
```
Comment (bocharsky-bw) [inline knpu/refresh-token.rst]: Remove extra spaces

File: knpu/refresh-token.rst [hunk 33]
```diff
@@ -148,17 +144,17 @@ value is today or earlier::
...
+```
 
 Next, let's iterate over each expiring token. To get a refresh token, we'll
-make an API request to the very-familiar ``/token`` endpoint. In fact, I'll
-start by copying the Guzzle API call from ``CoopOAuthController``::
+make an API request to the very-familiar `/token` endpoint. In fact, I'll
+start by copying the Guzzle API call from `CoopOAuthController`::
```
Comment (bocharsky-bw) [inline knpu/refresh-token.rst]: `:` and code block

File: knpu/refresh-token.rst [hunk 34]
```diff
@@ -184,9 +180,9 @@ start by copying the Guzzle API call from ``CoopOAuthController``::
     }
 
 
-Of course, we don't have a ``$code`` variable, but we *do* have the user's
-refresh token. Change ``grant_type`` to be ``refresh_token`` and replace
-the ``code`` parameter with the ``refresh_token``. We can also remove the ``redirect_uri``,
+Of course, we don't have a `$code` variable, but we *do* have the user's
+refresh token. Change `grant_type` to be `refresh_token` and replace
+the `code` parameter with the `refresh_token`. We can also remove the `redirect_uri`,
 which isn't needed with this grant type::
```
Comment (bocharsky-bw) [inline knpu/refresh-token.rst]: `:` and php code block

File: knpu/refresh-token.rst [hunk 35]
```diff
@@ -213,26 +209,26 @@ return any tokens expiring in the next month, which should be everyone::
 
 Now, try it by executing the script from the command line:
 
-.. code-block:: bash
-
-    $ php data/refresh_token.php
+```bash
```
Comment (bocharsky-bw) [inline knpu/refresh-token.rst]: `terminal`

File: knpu/security.rst [hunk 36]
```diff
@@ -35,7 +33,7 @@ that by generating a random string and storing it in the session::
         return $this->redirect($url);
     }
 
-Let's also add a ``die`` statement in the ``receiveAuthorizationCode`` function
+Let's also add a `die` statement in the `receiveAuthorizationCode` function
```
Comment (bocharsky-bw) [inline knpu/security.rst]: `receiveAuthorizationCode()`

File: knpu/security.rst [hunk 37]
```diff
@@ -45,10 +43,10 @@ that's executed after COOP redirects back to us::
...
 Log out and click to login via COOP. Of course, when we redirect to COOP,
-the new ``state`` parameter is there. Interestingly, after we authorize, COOP
-redirects back to us and *also* includes that exact ``state`` parameter.
+the new `state` parameter is there. Interestingly, after we authorize, COOP
+redirects back to us and *also* includes that exact `state` parameter.
 
-In ``receiveAuthorizationCode``, we just need to make sure that ``state``
+In `receiveAuthorizationCode`, we just need to make sure that `state`
 matches the string that we set in the session exactly. If it doesn't, let's
 render an error page: this could be an attack::
```
Comment (bocharsky-bw) [inline knpu/security.rst]: `:` and php code block

File: knpu/javascript-implicit-google-plus.rst [hunk 38]
```diff
@@ -23,30 +21,27 @@ a terrible idea.
 Instead, we need to look at one more grant type called Implicit. It's a lot
 like Authorization Code, but simpler.
 
-JavaScript OAuth with Google+
------------------------------
+## JavaScript OAuth with Google+
 
-To integrate with Google+, let's start by finding their `JavaScript Quick Start`_,
+To integrate with Google+, let's start by finding their [JavaScript Quick Start][JavaScript Quick Start],
 which is a little example app. If we follow the `Google+ Sign-In button`_,
```
Comment (bocharsky-bw) [inline knpu/javascript-implicit-google-plus.rst]: Missed link?

File: knpu/javascript-implicit-google-plus.rst [hunk 39]
```diff
@@ -91,169 +84,166 @@ fill these in for security purposes - a topic we'll cover later.
...
+for more information.
+```
 
 What we're trying to do here is *similar* to the step in the Authorization
 Code grant type where we originally redirect the user to the OAuth server.
 There are details we need to send to Google+, like our client id and the
 scopes we want.
 
-In fact, the ``gapi.auth`` object has `nice documentation`_ and the ``signIn``
+In fact, the `gapi.auth` object has [nice documentation][nice documentation] and the `signIn`
```
Comment (bocharsky-bw) [inline knpu/javascript-implicit-google-plus.rst]: `singIn()`

File: knpu/authorization-code.rst [hunk 40]
```diff
@@ -52,14 +50,13 @@ how many eggs each farmer has collected. The only missing piece is OAuth:
...
+## Redirecting to Authorize
 
 Before TopCluck can make an API request to COOP to count my eggs, I need
 to authorize it. On the homepage, there's already an Authorize link, which
 just prints a message right now.
 
-The code behind this URL lives in the ``src/OAuth2Demo/Client/Controllers/CoopOAuthController.php``
+The code behind this URL lives in the `src/OAuth2Demo/Client/Controllers/CoopOAuthController.php`
 file. You don't even need to understand how this works, just know that whatever
 we do here, shows up::
```
Comment (bocharsky-bw) [inline knpu/authorization-code.rst]: `::` -> `:` and add \`\`\`php for the next code block

File: knpu/authorization-code.rst [hunk 41]
```diff
@@ -73,8 +70,8 @@ we do here, shows up::
 
 The first step of the authorization code grant type is to redirect the user
 to a specific URL on COOP. From here the user will authorize our app. 
-According to `COOP's API Authentication page`_, we need to redirect
-the user to ``/authorize`` and send several query parameters.
+According to [COOP's API Authentication page][COOP's API Authentication page], we need to redirect
+the user to `/authorize` and send several query parameters.
 
 In our code, let's start building the URL::
```
Comment (bocharsky-bw) [inline knpu/authorization-code.rst]: the same...

File: knpu/authorization-code.rst [hunk 42]
```diff
@@ -93,29 +90,29 @@ In our code, let's start building the URL::
...
-Copy the ``client_id`` into our URL. Great! The last piece is the ``redirect_uri``,
+Copy the `client_id` into our URL. Great! The last piece is the `redirect_uri`,
 which is a URL on our site that COOP will send the user to after granting
 or denying our application access. We're going to do all kinds of important
 things once that happens.
 
-Let's set that URL to be ``/coop/oauth/handle``, which is just another page
+Let's set that URL to be `/coop/oauth/handle`, which is just another page
 that's printing a message. The code for this is right inside the same file,
 a little further down::
```
Comment (bocharsky-bw) [inline knpu/authorization-code.rst]: and here...

File: knpu/authorization-code.rst [hunk 43]
```diff
@@ -156,37 +153,35 @@ authorize URL to COOP, let's redirect the user to it::
...
 This query parameter is called the authorization code, and it's unique
 to this grant type. It's not an access token, which is really want we want,
 but it's the key to getting that. The authorization code is our temporary
 proof that the user said that our application can have an access token.
 
-Let's start by copying the code from the ``collect_eggs.php`` script and 
-pasting it here. Go ahead and change the ``client_id`` and ``client_secret`` 
+Let's start by copying the code from the `collect_eggs.php` script and 
+pasting it here. Go ahead and change the `client_id` and `client_secret` 
 to be from the new client or application we created for TopCluck::
```
Comment (bocharsky-bw) [inline knpu/authorization-code.rst]: you know... `:` and code block

File: knpu/authorization-code.rst [hunk 44]
```diff
@@ -215,12 +210,12 @@ to be from the new client or application we created for TopCluck::
...
-and ``redirect_uri``. I'm already retrieving the ``code`` query parameter, so 
-let's fill these in. Make sure to also change the ``grant_type`` to 
-``authorization_code`` like it describes in the docs. Finally, dump the
-``$responseBody`` to see if this request works::
+If we look back at the COOP API Authentication docs, we'll see that `/token`
+has 2 other parameters that are used with the authorization grant type: `code`
+and `redirect_uri`. I'm already retrieving the `code` query parameter, so 
+let's fill these in. Make sure to also change the `grant_type` to 
+`authorization_code` like it describes in the docs. Finally, dump the
+`$responseBody` to see if this request works::
```
Comment (bocharsky-bw) [inline knpu/authorization-code.rst]: `:`...

File: knpu/authorization-code.rst [hunk 45]
```diff
@@ -293,13 +288,12 @@ until this access token expires::
...
-----------------------
+## Using the Access Token
 
 Just like in our CRON script, let's use the access token to make an API request!
-One of the endpoints is ``/api/me``, which returns information about the user
+One of the endpoints is `/api/me`, which returns information about the user
 that is tied to the access token. Let's make a GET request to this endpoint,
-setting the access token on the ``Authorization`` header, just like we did
+setting the access token on the `Authorization` header, just like we did
 before::
```
Comment (bocharsky-bw) [inline knpu/authorization-code.rst]: `:`...

File: knpu/authorization-code2.rst [hunk 46]
```diff
@@ -30,14 +28,13 @@ In our app, we're going to store it in the database::
...
 
-Recording the Expires Time
-~~~~~~~~~~~~~~~~~~~~~~~~~~
+### Recording the Expires Time
 
-We can also store the time when the token will expire. I'll create a ``DateTime``
+We can also store the time when the token will expire. I'll create a `DateTime`
 object that represents the expiration time. We can check this
 later before trying to make API requests. If the token is expired, we'll
 need to send the user through the authorization process again::
```
Comment (bocharsky-bw) [inline knpu/authorization-code2.rst]: `:`...

File: knpu/authorization-code2.rst [hunk 47]
```diff
@@ -60,25 +57,24 @@ need to send the user through the authorization process again::
...
 
 This is what a canceled authorization looks like: no authorization code.
 
 Unfortunately, we can't just assume that the user authorized our application.
-As we've seen when this happens, the ``code`` query parameter will be missing, 
+As we've seen when this happens, the `code` query parameter will be missing, 
 but the OAuth server should include a few extra query parameters explaining what 
-went wrong. These are commonly called ``error`` and ``error_description``. Let's 
+went wrong. These are commonly called `error` and `error_description`. Let's 
 grab these and pass them into a template I've already prepared::
```
Comment (bocharsky-bw) [inline knpu/authorization-code2.rst]: `:`...

File: knpu/authorization-code2.rst [hunk 48]
```diff
@@ -140,12 +135,11 @@ can see what the problem was::
...
 
-Redirecting after Success
--------------------------
+## Redirecting after Success
 
-Until now, we've had an ugly ``die`` statement at the bottom of the code
+Until now, we've had an ugly `die` statement at the bottom of the code
 that handles the OAuth redirect. What you'll actually want to do here is
 redirect to some other page. Our work is done for now, so we want to help
 the user to continue on our site::
```
Comment (bocharsky-bw) [inline knpu/authorization-code2.rst]: `:`...

File: knpu/authorization-code2.rst [hunk 49]
```diff
@@ -162,63 +156,61 @@ In our application, this code simply redirects us to the homepage. And just
...
 
-Making the eggs-count API Request
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+### Making the eggs-count API Request
 
-Start by copying the ``/api/me`` code from ``CoopOAuthController``, and changing
-the method from ``get`` to ``post``, since the ``eggs-count`` endpoint requires
+Start by copying the `/api/me` code from `CoopOAuthController`, and changing
+the method from `get` to `post`, since the `eggs-count` endpoint requires
 POST::
```
Comment (bocharsky-bw) [inline knpu/authorization-code2.rst]: `:`...

File: knpu/authorization-code2.rst [hunk 50]
```diff
@@ -246,9 +238,9 @@ POST::
         }
     }
 
-The endpoint we want to hit now is ``/api/USER_ID/eggs-count``. Fortunately,
+The endpoint we want to hit now is `/api/USER_ID/eggs-count`. Fortunately,
 we've already saved the COOP user id and access token for the currently logged-in
-user to the database. Get that data by using our app's ``$this->getLoggedInUser()``
+user to the database. Get that data by using our app's `$this->getLoggedInUser()`
 method and update the URL::
```
Comment (bocharsky-bw) [inline knpu/authorization-code2.rst]: `:`...

File: knpu/authorization-code2.rst [hunk 51]
```diff
@@ -307,12 +299,11 @@ right, Farmer Brent's egg count isn't going up. Let's go to COOP and
...
 
-All the Things that can Go Wrong
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+### All the Things that can Go Wrong
 
 The "Count Eggs" page we created works great, but we're not handling any
 of the things that might go wrong. First, we're hiding its link, but what
-if a user somehow ends up on the page without a ``coopUserId`` or ``coopAccessToken``?
+if a user somehow ends up on the page without a `coopUserId` or `coopAccessToken`?
 Let's code for this case::
```
Comment (bocharsky-bw) [inline knpu/authorization-code2.rst]: `:`...

File: knpu/client-credentials.rst [hunk 52]
```diff
@@ -129,27 +125,26 @@ If you visit the application you created earlier, you'll see a nice
...
-    GET /api/barn-unlock HTTP/1.1
-    Host: coop.apps.knpuniversity.com
-    Authorization: Bearer ACCESSTOKENHERE
+```text
+GET /api/barn-unlock HTTP/1.1
+Host: coop.apps.knpuniversity.com
+Authorization: Bearer ACCESSTOKENHERE
+```
 
 Update the script to send this header::
```
Comment (bocharsky-bw) [inline knpu/client-credentials.rst]: `:`

**Comment by bocharsky-bw** [comment]
@larzuk91 a tiny advice from me: I suggest to find a repeating problem, fix all its cases, and then, when you push changes, GitHub hides all my affected comments, so you'll be easier to see what is left :)

File: knpu/authorization-code.rst [hunk 53]
```diff
@@ -10,38 +9,37 @@ Then it hits him! The COOP API has an endpoint to see how many eggs have
...
----------------
+## Starting up FCL
 
 Let's check out the FCL app, which Brent has already started building. It
-lives in the  ``client/`` directory of the code download. I'll use the built-in
+lives in the  `client/` directory of the code download. I'll use the built-in
 PHP web server to run this site:
 
-.. code-block:: bash
+```bash
```
Comment (larzuk91) [inline knpu/authorization-code.rst]: Ohh, so all bash code blocks should be changed to terminal ?

**Comment by larzuk91** [comment]
@bocharsky-bw  Ohh great, thanks!
Looks like I forgot completely to convert double semicolon to single one ;)

File: knpu/authorization-code.rst (hunk 40 above)
Comment (larzuk91) [inline knpu/authorization-code.rst]: Ohhh I missed all this codeblocks because there is not a codeblock definition on the file =/

**Comment by larzuk91** [comment]
Ok, this is finished! :D
@weaverryan we can deploy this one

File: knpu/authorization-code.md [hunk 54]
```diff
@@ -124,12 +129,14 @@ a little further down::
         // equivalent to $_GET['code']
         $code = $request->get('code');
 
-        die('Implement this in CoopOAuthController::receiveAuthorizationCode');
+        die('Implement this in CoopOAuthController:receiveAuthorizationCode');
```
Comment (bocharsky-bw) [inline knpu/authorization-code.md]: @larzuk91 Gotcha! Auto-replace is dangerous thing, but PhpStorm allows you to do auto-replace with reviewing, which should prevent this cases ;)

**Comment by bocharsky-bw** [comment]
@larzuk91 GREAT! Good work! I just found one minor issue which I fixed by myself, so I'm going to merge it now.

What about avoiding this PR from being so big - you can create a PR for each file, but anyway then we have to merge those branches between each other, and only then merge all that to master. So don't worry that this PR is big, it's a normal thing for this kind of stuff.

And thanks for your help on it!

File: knpu/authorization-code.md (hunk 54 above)
Comment (larzuk91) [inline knpu/authorization-code.md]: Hmm, weird thing, I specified that only replace if it ends with break line
for some reason I missed this one, nice catch btw ;)

**Pull request merged**